### PR TITLE
Oil 203 config execute button

### PR DESCRIPTION
**Pull request by subsend**

### CODE REVIEW

File: src/scripts/inputFields/input_container.jsx [hunk 1]
```diff
@@ -25,46 +25,49 @@ class InputContainer extends React.Component {
 
     Object.keys(configElements).map((fieldName) => {
       const fullFieldName = (this.props.baseAttribute ? this.props.baseAttribute + '.' : '') + fieldName;
+      const field = configElements[fieldName];
+      const help_text = this.determineHelpText(field);
```
Comment (phogel) [inline src/scripts/inputFields/input_container.jsx]: maybe call this variable `helpText` for consistency?

File: src/scripts/oil-configuration-manager.jsx [hunk 2]
```diff
@@ -15,6 +15,7 @@ class OilConfigurationManager extends React.Component {
 
   build() {
     ReactDOM.render(<ManagerView config={this.state.config}/>, document.querySelector("#oil-configuration-manager-container"));
+    console.log(`config: ${JSON.stringify(this.state.config, null, 2)}`);
```
Comment (phogel) [inline src/scripts/oil-configuration-manager.jsx]: Needs to be deleted

File: src/scripts/inputFields/string_input.jsx [hunk 3]
```diff
@@ -12,13 +18,36 @@ class StringInput extends Input {
...
+      attribute: this.props.attribute,
+      value: newValue,
+      helpText: this.props.helpText
+    };
+
+    if (newValue.match(this.props.pattern)) {
+      this.setState({
+        classes: ''
+      });
+      this.props.emitter.emit('validField', options);
```
Comment (ltparis2018) [inline src/scripts/inputFields/string_input.jsx]: Please, avoid this. Now we have two ways of communication with the JSON view within the input fields - handleUpdate() and emitter. Better than this mixture would be one of the following variants:
- always use the emitter and remove the handleUpdate() methods (not my favorite)
- leave emitter in configurators and extend handleUpdate() or create a new handleValidationResult() method.

File: src/scripts/inputFields/input_container.jsx [hunk 4]
```diff
@@ -83,13 +86,15 @@ class InputContainer extends React.Component {
...
+  fieldForString(attribute, value, helpText, pattern) {
     return (
       <StringInput key={attribute}
                    handleUpdate={this.props.handleUpdate}
                    attribute={attribute}
                    value={value}
-                   helpText={helpText}/>
+                   helpText={helpText}
+                   pattern={pattern}
+                   emitter={this.props.emitter}/>
```
Comment (ltparis2018) [inline src/scripts/inputFields/input_container.jsx]: Here, the emitter is propagated to StringInput and ObjectInput. But the ListInput implementation that creates StringInput and ObjectInput instances too, don't do this. This is inconsistent! Please, see my other comments to this point for an alternative...

File: src/components/configurationManager.html [hunk 5]
```diff
@@ -1,10 +1,10 @@
-<div id="oil-configuration-manager-container" class="widget-section" style="display: block">
+<div id="oil-configuration-manager-container" class="widget-section">
   
 </div>
 <script type="text/javascript">
 function showConfigurationManager() {
   new OilConfigurationManager({
-    config: window.AS_OIL.CONFIG
+    config: JSON.parse($('#oil-configuration').value[0].text)
```
[the remark this answers is not in the document]
Comment (ltparis2018) [reply src/components/configurationManager.html]: Done.

**Pull request merged**